doc/using_the_march_exoskeleton/error_codes.rst

```diff
@@ -25,10 +25,117 @@ without having to repeat the whole error description.
+----------------------------------
+Invalid IU position command.
+
+**Cause:** The position you are trying to actuate to is not a valid target within the range of the joint.
+
+**Fix:** Check that the IU limits defined in the :hardware-interface:`robots yaml <march_hardware_builder/robots>`
+under ``joints/<joint_name>/imotioncube/encoder`` match the actual limits of the joint.
+
+
+.. _e102:
+
+``E102``: Encoder reset
+-----------------------
+An encoder has reset and reads an incorrect value.
+
+**Cause:** The cause is still unknown.
+
+**Fix:** This error happens sometimes during startup and the only thing you
+can do is try again. This error will probably be removed once this issue has
+been fixed.
+
+
+.. _e103:
+
+``E103``: Outside hard limits
+-----------------------------
+A joint is outside its defined hard limits.
+
+**Cause:** A joint was found to be outside its defined hard limits in the
+:hardware-interface:`robots yaml <march_hardware_builder/robots>`.
+
+**Fix:** Check that the IU limits defined in the :hardware-interface:`robots yaml <march_hardware_builder/robots>`
+under ``joints/<joint_name>/imotioncube/encoder`` match the actual limits of the joint and recalibrate if necessary.
+
+
+.. _e104:
+
+``E104``: Target exceeds max difference
+---------------------------------------
+The target position exceeds the max allowed difference from the current position.
+
+**Cause:** The given target position is greater than the maximum allowed difference
+between target and actual position. This could cause dangerous situations if
+you were to actuate a long distance in a short amount of time.
+
+**Fix:** This is a safety measure and most of time indicates that the joints
+are behind their trajectory path.
+
+
+.. _e105:
+
+``E105``: Target torque exceeds max torque
+------------------------------------------
+The target torque exceeds the max allowed torque
+
+**Cause:** The maximum given target torque is greater than the joints
+can actuate. This could cause dangerous situations where the joints could be
+negatively affected.
+
+**Fix:** This is a safety measure and most of time indicates that the joints
+are behind their trajectory path.
+
+
+.. _e106:
+
+``E106``: PDO object not defined
+--------------------------------
+The to be added PDO object was not defined.
+
+**Cause:** The PDO that is being mapped does not have a defined address and size.
+
+**Fix:** Check that the PDO is defined in the ``all_objects`` map in the
+:hardware-interface:`PDOmap <march_hardware/src/PDOmap.cpp>` class.
+
+
+.. _e107:
+
+``E107``: PDO register overflow
+-------------------------------
+The PDO map could not fit within the registers
+
+**Cause:** There exist 4 registers with 64 bits each to be filled with PDO maps.
+This error indicates that the added PDO exceeded 4*64=256 bits.
+
+**Fix:** Check that you actually need all the mapped PDOs and remove some that
+you are not using until you are at or below the 256 bits.
+
+
```

Comment:
Is writing initial settings / invalid input configuration not also an error?

Reply:
It is not an error I added to the `IMotionCube` class, so I will create it there and then add it here